File: tutorial/Ex05b.java
// Ex5b.java
// Several rotatable multi-sprite actors like PacMan ghosts

import ch.aplu.jgamegrid.*;

public class Ex05b extends GameGrid
{
  public Ex05b()
  {
    super(800, 600, 1, null, true);
    setSimulationPeriod(20);
    setBgColor(new java.awt.Color(100, 100, 100));
    Ghost[] ghost = new Ghost[7];
    for (int i = 0; i < 7; i++)
    {
      ghost[i] = new Ghost();
      Location startLocation = new Location(310 + 30 * i, 100);
      int startDirection = 66 + 66 * i;
      addActor(ghost[i], startLocation, startDirection);
    }
    show();
  }

  public static void main(String[] args)
  {
    new Ex05b();
  }
}

File: python/GameGridEx/Ex05b.py
# Ex05b.py

from ch.aplu.jgamegrid import GameGrid, Actor, Location, X11Color

# --------------------- class Ghost --------------------------
class Ghost(Actor):
   def __init__(self):
      Actor.__init__(self, "sprites/ghost.gif", 4)
   
   def act(self):
      if self.getNbCycles() % 5 == 0:
         self.showNextSprite()

      pt = gg.toPoint(self.getNextMoveLocation())
      dir = self.getDirection()

      if pt.x < 0 or pt.x > gg.getPgWidth():
        self.setDirection(180 - dir)

      if pt.y < 0 or pt.y > gg.getPgHeight():
        self.setDirection(360 - dir)

      self.move();
 
# --------------------- main ---------------------------------
gg = GameGrid(800, 600, 1, None, True)
gg.setSimulationPeriod(20)
gg.setBgColor(X11Color("darkgray"))
for i in range(7):
   ghost = Ghost()
   startLocation = Location(310 + 30 * i, 100)
   startDirection = 66 + 66 * i
   gg.addActor(ghost, startLocation, startDirection)
gg.show();


